aion_pool/src/Miningcore/Blockchain/Aion/AionRewardsCalculator.cs

```diff
@@ -27,5 +41,13 @@ class AionRewardsCalculator
                 return (decimal) blockReward / magnitude;
             }
         }
+
+        public decimal calculateRewardWithTimeSpan(long timeSpan) {
+            if (timeSpan == 0) {
+                return 0;
+            }
+
+            return rewardsAdjustTable[timeSpan > capping ? capping - 1 : timeSpan - 1];
```

Comment:
Any reason for timespan and capping - 1? Would that not possibly give us an out of index error (since we start from index 1 assuming that the minimum block delta is 1 second).

Reply:
Because the array index starts from zero, So actually the rewardsAdjustTable[0] is storing the rewards for 1 sec timespan.
rewardsAdjustTable[69] is the capping of the rewards.

Reply:
Not sure I follow, wouldn't rewardsAdjustTable[0] be storing the 0th second timestamp, Following the formula listed above, isn't the value calculated at i=0 (rewardsAdjustTable[0]) the intercept of the linear curve?

Reply:
Ahh I gotcha, there is a +1 added to the delta, okay LGTM, but we should add a comment here to make it clear